app/controllers/uploads_controller.rb

```diff
@@ -317,14 +316,22 @@ def self.create_upload(
     tempfile&.close!
   end
 
+  def self.max_attachment_size_for_user(user)
+    if user.id == Discourse::SYSTEM_USER_ID && !SiteSetting.system_user_max_attachment_size_kb.zero?
+      SiteSetting.system_user_max_attachment_size_kb
+    else
+      SiteSetting.max_attachment_size_kb
+    end
+  end
```

Comment:
We can probably make this a private instance method in stead of a public class method.